src/biokbase/narrative/widgetmanager.py

```diff
@@ -396,7 +396,7 @@ def infer_upas(self, widget_name, params):
         obj_ref_list = list()   # list of tuples, but second is a list of upas
         ws = None
         for param in params.keys():
-            if param in param_to_context:
+            if param in param_to_context and params.get(param) is not None:
```

Comment:
Are there cases where `None` is a valid value for a param? For example, I've used `False, None, True` as a ternary.

Reply:
Yes, `None` can be a valid value. But in this case, that block of code is trying to map values onto workspace object identifier. If it sees `None`, it should just accept that there's nothing there and not try to turn it into an object name/id.

Reply:
The docs for the method confused me a bit - now I look at the code it's clear that most of the params are going to be ignored.